fastf1/core.py

```diff
@@ -2844,18 +2844,32 @@ def pick_fastest(self, only_by_time: bool = False) -> "Lap":
         Returns:
             instance of :class:`Lap`
         """
+        # TODO: Deprecate returning empty lap object when there is no lap
+        # that matches definion
         if only_by_time:
             laps = self  # all laps
         else:
             # select only laps marked as personal fastest
-            laps = self.loc[self['IsPersonalBest'] == True]  # noqa: E712 comparison with True
```

Comment:
The noqa is added in this [commit](https://github.com/theOehrly/Fast-F1/commit/23597f04141c574451932552a42ba7244278d3ee)

Thought it looked strange. Why was it changed originally?

Reply:
Just triggered the doctest failure that I assume was referred to in the original commit. I will see if I can there is a simple native Pandas solution before reverting